scripts/renderers.py

```diff
@@ -20,31 +20,66 @@ class CsvRenderer():
     Used to output toggles+annotations data as CSS
     """
 
-    def transform_toggle_data_for_csv(self, envs_data):
+    def render_csv_report(self, idas, file_path="report.csv", toggle_types=None, header=None, summarize=False):
         """
-        Retrieve list of individual toggle datums from envs_data
+        takes data, processes it, and outputs it in csv form
+        """
+        if summarize:
+            toggle_info_structured_dicts = self.summarize_data(idas)
+        else:
+            toggle_info_structured_dicts = self.output_full_data(idas)
+
+        toggles_info_flattened_dicts = self.add_info_source_to_dict_keys(toggle_info_structured_dicts)
 
-        envs_data is a dict with a complex structure of environments, idas and toggles by toggle type.
-        Return a flattened list for each toggle in a specific environment in preparation for csv output.
+        sorted_toggles_dicts = self.filter_and_sort_toggles(toggles_info_flattened_dicts, toggle_types)
+        header = self.get_sorted_headers_from_toggles(sorted_toggles_dicts, header)
+        self.write_csv(file_path, sorted_toggles_dicts, header)
+
+    def output_full_data(self, idas):
         """
-        toggles_data = []
-        for env, idas in envs_data.items():
-            for ida_name, ida in idas.items():
-                for toggle_type, toggles in ida.toggles.items():
-                    for toggle_name, toggle in toggles.items():
-                        data_dict = toggle.full_data()
-                        data_dict["toggle_type"] = toggle_type
-                        # In case you want the report to call the ida by a different ida_name
-                        # example: lms should be called edxapp in report
-                        data_dict["ida_name"] = ida.configuration.get("rename", ida_name)
-                        data_dict["env_name"] = env
-                        toggles_data.append(data_dict)
-        return toggles_data
-
-    def filter_and_sort_toggles(self, toggles_data, toggle_type_filter=None):
+        Return full data about toggles.
+
+        Essentially, one row per toggle state (one toggle with multiple states will have multiple rows)
+        """
+        output = []
+        for ida in idas.values():
+            output.extend(ida.get_full_report())
+        return output
+
+    def summarize_data(self, idas):
+        """
+        Returns only subset containing the essential information
+        Essentially, one row per toggle
+        """
+        output = []
+        for ida in idas.values():
+            output.extend(ida.get_toggles_data_summary())
+        return output
```

Comment:
I'm not sure how many reports we have at this point.  Is it just these two?

I think full_data, or verbose, includes one row for the toggle state in any specific environment.  When multiple environments are included, the toggle annotations will be repeated for each of these rows.  Correct?

For the summary, this data is collapsed to a singe row, with state data per environment in different columns, and a column to summarize diffs across environments.  Right?

I think a longer english description would help in some of the function docstrings, as well as the docs.  I'm not clear on the rendering difference between these two, and whether the renderer needs these two methods, or whether you just have two different reports (and data) that are passed to the renderer.